src/jit/gentree.cpp

```diff
@@ -14079,62 +14128,31 @@ bool Compiler::gtNodeHasSideEffects(GenTreePtr tree, unsigned flags)
     {
         if (tree->OperGet() == GT_CALL)
         {
-            // Generally all GT_CALL nodes are considered to have side-effects.
-            // But we may have a helper call that doesn't have any important side effects.
-            //
-            if (tree->gtCall.gtCallType == CT_HELPER)
+            GenTreeCall* const call             = tree->AsCall();
+            const bool         ignoreExceptions = (flags & GTF_EXCEPT) == 0;
+            const bool         ignoreCctors     = (flags & GTF_IS_IN_CSE) != 0; // We can CSE helpers that run cctors.
```

Comment:
We could use a function header for this method describing that the caller passes in `flags` that control the behavior of this method.   It seems a bit odd to me to use the  `flags`  argument on this way.

Alternatively we could pass in two values : 1) ignoreExceptions  and 2 ignoreCctors